Lazy load entities on client initialization

rbarman opened the pull request:
Lazy load entities on client initialization

### Description

Lazy load entities (players, teams, stadia, managers, referees) on client initialization. This is related to https://github.com/American-Soccer-Analysis/itscalledsoccer/issues/169

- Add `lazy_load` parameter to the `AmericanSoccerAnalysis` constructor. If it is set to true, then all entities (players, teams, stadia, managers, referees) will be initialized to None, else they will be initialized.
- All of the entity getter functions (`get_stadia`, `get_teams`, etc) load the entity data if `self.<entity>` is None. 

### Checklist

- [X] Code compiles correctly
- [X] Created tests which fail without the change (if applicable)
- [X] All lint and unit tests passing
- [X] Extended the README / documentation, if necessary

### Additional Comments

I also updated `tests\features\steps\common_steps.py` to read the mock payloads with `encoding='utf-8'`. The behave tests were failing for me without explicit utf-8 encoding.

bgrnwd commented on itscalledsoccer/client.py:
```diff
@@ -46,11 +47,22 @@ def __init__(
+        self.lazy_load = lazy_load
+
+        if self.lazy_load:
+            print("Lazy loading enabled. Initializing client without entity data.")
```
```suggestion
            self.LOGGER.info("Lazy loading enabled. Initializing client without entity data.")
```
We should use the `LOGGER` instead of `print`

bgrnwd reviewed (changes requested):
Thanks for the PR!

It looks like we have some `mypy` failures that we need to resolve. You should be able to install the dev dependencies by running `pip install ".[dev]"` in the root directory of the repo. After that, you can run `mypy itscalledsoccer`

rbarman commented:
Thanks for the feedback @bgrnwd!

I updated the code to address the mypy issues and replaced print with the logger. Can you take a look at my changes?

bgrnwd commented:
@rbarman Looks good now 👍 I'll get a new version published after #172 is merged in
